https://trac.macports.org/ticket/50694
https://sourceforge.net/p/wxsvg/discussion/424987/thread/576bf3a0/

PIX_FMT_RGB24 has already been replaced by AV_PIX_FMT_RGB24 upstream
    http://wxsvg.cvs.sourceforge.net/viewvc/wxsvg/wxsvg/src/mediadec_ffmpeg.cpp?r1=1.32&r2=1.33
The other patch is optional for now. The compiler throws deprecation warnings only.

--- src/mediadec_ffmpeg.cpp.orig
+++ src/mediadec_ffmpeg.cpp
@@ -263,9 +263,9 @@ wxImage wxFfmpegMediaDecoder::GetNextFrame() {
 			avcodec_decode_video2(m_codecCtx, m_frame, &frameFinished, &packet);
 			if (frameFinished) {
 				SwsContext* imgConvertCtx = sws_getContext(m_codecCtx->width, m_codecCtx->height, m_codecCtx->pix_fmt,
-						m_width, m_height, PIX_FMT_RGB24, SWS_BICUBIC, NULL, NULL, NULL);
+						m_width, m_height, AV_PIX_FMT_RGB24, SWS_BICUBIC, NULL, NULL, NULL);
 				if (imgConvertCtx == NULL) {
-					av_free_packet(&packet);
+					av_packet_unref(&packet);
 					return wxImage();
 				}
 
@@ -273,13 +273,13 @@ wxImage wxFfmpegMediaDecoder::GetNextFrame() {
 				uint8_t *rgbSrc[3] = { img.GetData(), NULL, NULL };
 				int rgbStride[3] = { 3 * m_width, 0, 0 };
 				sws_scale(imgConvertCtx, m_frame->data, m_frame->linesize, 0, m_codecCtx->height, rgbSrc, rgbStride);
-				av_free_packet(&packet);
+				av_packet_unref(&packet);
 				sws_freeContext(imgConvertCtx);
 				return img;
 			}
 		}
 		// free the packet that was allocated by av_read_frame
-		av_free_packet(&packet);
+		av_packet_unref(&packet);
 	}
 	return wxImage();
 }
